azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -257,78 +252,75 @@ public String internalFqdn() {
     }
 
     @Override
-    public String subnetId() {
-        return this.primaryIpConfiguration().subnet().id();
+    public Network primaryNetwork() throws CloudException, IOException {
+        return this.primaryIpConfiguration().network();
     }
 
     @Override
-    public Network network() throws CloudException, IOException {
-        String id = subnetId();
-        return this.networks.get(ResourceUtils.groupFromResourceId(id), ResourceUtils.extractFromResourceId(id, "virtualNetworks"));
+    public String primaryPrivateIp() {
+        return this.primaryIpConfiguration().privateIp();
     }
 
     @Override
-    public String privateIp() {
-        return this.primaryIpConfiguration().privateIPAddress();
+    public String primaryPrivateIpAllocationMethod() {
+        return this.primaryIpConfiguration().privateIpAllocationMethod();
     }
 
     @Override
-    public String privateIpAllocationMethod() {
-        return this.primaryIpConfiguration().privateIPAllocationMethod();
+    public List<NicIpConfiguration> ipConfigurations() {
+        return Collections.unmodifiableList(this.nicIpConfigurations);
     }
 
-
     /**************************************************.
      * CreatableImpl::createResource
      **************************************************/
 
     @Override
     protected void createResource() throws Exception {
-        this.primaryIpConfiguration().setSubnet(subnetToAssociate());
-        this.primaryIpConfiguration().setPublicIPAddress(publicIpToAssociate());
-        ServiceResponse<NetworkInterfaceInner> response = this.client.createOrUpdate(this.resourceGroupName(), this.nicName, this.inner());
+        for (NicIpConfiguration ipConfig : this.nicIpConfigurations) {
+            ((NicIpConfigurationImpl)ipConfig).ensureConfiguration();
+        }
+
+        ServiceResponse<NetworkInterfaceInner> response = this.client.createOrUpdate(this.resourceGroupName(),
+                this.nicName,
+                this.inner());
         this.setInner(response.getBody());
+        initializeNicIpConfigurations();
     }
 
     /**************************************************.
      * Helper methods
      **************************************************/
 
     /**
-     * @param prefix the prefix
-     * @return a random value (derived from the resource and resource group name) with the given prefix
-     */
-    private String nameWithPrefix(String prefix) {
-        return prefix + "-" + this.randomId + "-" + this.resourceGroupName();
-    }
-
-    /**
      * @return the primary IP configuration of the network interface
      */
-    private NetworkInterfaceIPConfiguration primaryIpConfiguration() {
-        if (this.inner().ipConfigurations().size() == 0) {
-            this.inner().ipConfigurations().add(new NetworkInterfaceIPConfiguration());
-            this.inner().ipConfigurations().get(0).setName("primary-nic-config");
+    private NicIpConfigurationImpl primaryIpConfiguration() {
```

Comment:
Somehow the comments became outdated.
Since this method is not throwing `CloudException` & `IOException`, the other ones that call this shouldn't declare `throws` either?